### PR TITLE
Add profiles to CLI

### PR DESCRIPTION
**Pull request by henneberger**
Adds the ability to specify profiles through the cli "-p" switch. It will look for a folder of that name and merge all of the configs.

### CODE REVIEW

File: sqrl-tools/sqrl-cli/src/main/java/com/datasqrl/cmd/DiscoverCommand.java [hunk 1]
```diff
@@ -67,8 +76,11 @@ protected void runCommand(ErrorCollector errors) throws IOException {
       statistics = false;
     }
 
-    SqrlConfig config = PackagerUtil.getOrCreateDefaultConfiguration(root, errors,
-                ()-> PackagerUtil.createEmbeddedConfig(root.rootDir, errors));
+    //check package json, or use embedded config
+    SqrlConfig config = Packager.findPackageFile(root.rootDir, this.root.packageFiles)
+        .map(p -> SqrlConfigCommons.fromFiles(errors, p))
+        .orElseGet(() -> createEmbeddedConfig(errors));
```
Comment (henneberger) [inline sqrl-tools/sqrl-cli/src/main/java/com/datasqrl/cmd/DiscoverCommand.java:82]: @mbroecheler Should discovery resolve profiles? I see arguments for both ways.

File: sqrl-tools/sqrl-cli/src/main/java/com/datasqrl/cmd/AbstractCompilerCommand.java [hunk 2]
```diff
@@ -3,121 +3,103 @@
...
+      description = "An alternative set of configuration values which override the default package.json")
+  protected String[] profiles = new String[0];
+
+  @SneakyThrows
+  public void execute(ErrorCollector errors) {
+    Repository repository = createRepository(errors);
 
-  public void runCommand(ErrorCollector errors) {
-    setupTargetDir();
+    PackageBootstrap packageBootstrap = new PackageBootstrap(root.rootDir,
```
Comment (mbroecheler) [inline sqrl-tools/sqrl-cli/src/main/java/com/datasqrl/cmd/AbstractCompilerCommand.java:65]: Nice, I like the breakdown into bootstrap -> config -> packager -> build directory.

File: sqrl-tools/sqrl-cli/src/main/java/com/datasqrl/cmd/DiscoverCommand.java (hunk 1 above)
Comment (mbroecheler) [reply sqrl-tools/sqrl-cli/src/main/java/com/datasqrl/cmd/DiscoverCommand.java:82]: I think of discover as something that folks run locally to get their data packages without much work. Hence, I don't think we need the full flexibility of profiles. At least not yet.

File: sqrl-tools/sqrl-cli/src/test/resources/snapshots/com/datasqrl/cmd/PackageBootstrapTest/testRetailPackaging.txt [hunk 3]
```diff
@@ -232,7 +237,7 @@ package.json
   },
   "script" : {
     "graphql" : "c360-full-graphqlv1/schema.graphqls",
-    "main" : "c360-full.sqrl"
+    "main" : "c360-full-graphqlv1/schema.graphqls"
```
Comment (mbroecheler) [inline sqrl-tools/sqrl-cli/src/test/resources/snapshots/com/datasqrl/cmd/PackageBootstrapTest/testRetailPackaging.txt:240]: This doesn't look right.

File: sqrl-tools/sqrl-cli/src/test/resources/snapshots/com/datasqrl/cmd/PackageBootstrapTest/testRemoteRepository.txt [hunk 4]
```diff
@@ -1,5 +1,10 @@
 >>>src/test/resources/dependency/main1.sqrl-src/test/resources/dependency/packageWODependencies.json-dir
+build
 build.gradle
+build/datasqrl
```
Comment (mbroecheler) [inline sqrl-tools/sqrl-cli/src/test/resources/snapshots/com/datasqrl/cmd/PackageBootstrapTest/testRemoteRepository.txt:4]: why is there a build folder with the same nested folder structure in here?

File: sqrl-tools/sqrl-cli/src/test/resources/snapshots/com/datasqrl/cmd/PackageBootstrapTest/testRetailPackaging.txt [hunk 5]
```diff
@@ -16,6 +16,7 @@ ecommerce-data/customer.schema.yml
 ecommerce-data/customer.table.json
 ecommerce-data/orders.schema.yml
 ecommerce-data/orders.table.json
+ecommerce-data/package.json
```
Comment (mbroecheler) [inline sqrl-tools/sqrl-cli/src/test/resources/snapshots/com/datasqrl/cmd/PackageBootstrapTest/testRetailPackaging.txt:19]: Where do these package.json come from?

File: sqrl-tools/sqrl-packager/src/main/java/com/datasqrl/packager/Packager.java [hunk 6]
```diff
@@ -251,29 +218,91 @@ private boolean retrieveDependency(Path buildDir, NamePath packagePath, Dependen
...
+        return Optional.of(List.of(defaultPkg));
+      } else {
+        return Optional.empty();
+      }
+    } else {
+      return Optional.of(packageFiles);
+    }
+  }
+
+  public static SqrlConfig createDockerConfig(ErrorCollector errors) {
```
Comment (mbroecheler) [inline sqrl-tools/sqrl-packager/src/main/java/com/datasqrl/packager/Packager.java:265]: Why do we still need to hard-code this in here? Is it possible to make this a profile that we ship with DataSQRL?

**Review by mbroecheler — COMMENTED** [review]
This looks great in general. I added a couple of questions that I added inline. And another one: How can we test this well enough for release by end of week?

File: sqrl-tools/sqrl-packager/src/main/java/com/datasqrl/packager/Packager.java (hunk 6 above)
Comment (henneberger) [reply sqrl-tools/sqrl-packager/src/main/java/com/datasqrl/packager/Packager.java:265]: this is 'default' config for the compile and validate targets when no other package.json is specified. It's awkwardly named, but we can further modularize it out in the future.

File: sqrl-tools/sqrl-cli/src/test/resources/snapshots/com/datasqrl/cmd/PackageBootstrapTest/testRetailPackaging.txt (hunk 5 above)
Comment (henneberger) [reply sqrl-tools/sqrl-cli/src/test/resources/snapshots/com/datasqrl/cmd/PackageBootstrapTest/testRetailPackaging.txt:19]: The are in the examples dir

**Pull request merged**